catalog/nato/stanag4559/nato-stanag4559-source/src/main/java/org/codice/alliance/nato/stanag4559/source/Stanag4559Source.java

```diff
@@ -0,0 +1,942 @@
+        initOrb();
+    }
+
+    public Stanag4559Source() {
+        initOrb();
+        scheduler = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    public void init() {
+        createClientFactory();
+        initCorbaClient();
+        setupAvailabilityPoll();
+    }
+
+    private void createClientFactory() {
+        if (StringUtils.isNotBlank(cxfUsername) && StringUtils.isNotBlank(cxfPassword)) {
+            factory = new SecureCxfClientFactory(iorUrl,
+                    Stanag4559.class,
+                    null,
+                    null,
+                    true,
+                    false,
+                    null,
+                    null,
+                    cxfUsername,
+                    cxfPassword);
+        } else {
+            factory = new SecureCxfClientFactory(iorUrl,
+                    Stanag4559.class,
+                    null,
+                    null,
+                    true,
+                    false,
+                    null,
+                    null);
+        }
+    }
+
+    /**
+     * Initializes the Corba Client ORB and gets the mandatory interfaces that are required for a
+     * STANAG 4559 complaint Federated Source, also queries the source for the queryable attributes
+     * and views that it provides.
+     */
+    private void initCorbaClient() {
+        getIorFileFromSource();
+        initLibrary();
+        setSourceDescription();
+        initMandatoryManagers();
+        initServerViews();
+        initQueryableAttributes();
+        initSortableAndResultAttributes();
+        setFilterDelegate();
+    }
+
+    /**
+     * Uses the SecureClientCxfFactory to obtain the IOR file from the provided URL via HTTP(S).
+     */
+    private void getIorFileFromSource() {
+        Stanag4559 stanag4559 = factory.getClient();
+
+        try (InputStream inputStream = stanag4559.getIorFile()) {
+            iorString = IOUtils.toString(inputStream, UTF8);
+        } catch (IOException e) {
+            LOGGER.error("{} : Unable to process IOR String.", id);
+        }
+
+        if (StringUtils.isNotBlank(iorString)) {
+            LOGGER.debug("{} : Successfully obtained IOR file from {}", getId(), iorUrl);
+        } else {
+            LOGGER.error("{} : Received an empty or null IOR String.", id);
+        }
+    }
+
+    /**
+     * Initializes the Corba ORB with no additional arguments
+     */
+    private void initOrb() {
+        orb = org.omg.CORBA.ORB.init(new String[0], null);
+        if (orb != null) {
+            LOGGER.debug("{} : Successfully initialized CORBA orb.", getId());
+        } else {
+            LOGGER.error("{} : Unable to initialize CORBA orb.", getId());
+        }
+    }
+
+    /**
+     * Initializes the Root STANAG 4559 Library Interface
+     */
+    private void initLibrary() {
+        org.omg.CORBA.Object obj = orb.string_to_object(iorString);
+        library = LibraryHelper.narrow(obj);
+        if (library == null) {
```

Comment:
Might want to use the same null-checking pattern in each of these `init()` methods, i.e., the condition in the `if` statement should either check for null or not null. It makes the code easier to read.